patches/minecraft/net/minecraft/client/gui/GuiScreen.java.patch

```diff
@@ -1,6 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/client/gui/GuiScreen.java
 +++ ../src-work/minecraft/net/minecraft/client/gui/GuiScreen.java
-@@ -147,7 +147,10 @@
+@@ -63,6 +63,7 @@
+     private long field_146288_g;
+     private int field_146298_h;
+     private URI field_175286_t;
++    protected boolean keyHandled, mouseHandled; // Forge: allow canceling key and mouse Post events from handleMouseInput and handleKeyboardInput
```

Comment:
The goal is to minimize the complexity of the patch and make it easiest to understand WHILE keeping it as small as possible. This one could go either way and be the same. As its just field declarations and are not complex in any way.